### PR TITLE
Adding in the sync related models for the sync event webhook.

### PR DESCRIPTION
**Pull request by ChrisEby**
Used https://mandrill.zendesk.com/hc/en-us/articles/205583297-Sync-Event-Webhook-format for the format to accept sync requests.

### CODE REVIEW

**Comment by feinoujc** [comment]
Thanks for the contribution! you can disregard the appveyor build, it's not working for PR's at the moment.

**Comment by feinoujc** [comment]
I looked at the actual json samples using requestb.in and the docs might not be correct here :confused: 

for blacklist the details are in a property called `reject` but whitelist are in a property called `entry`

```json
[
    {
        "action": "add",
        "reject": {
            "created_at": "2014-01-15 11:32:19",
            "detail": "Example detail",
            "email": "example.webhook@mandrillapp.com",
            "expired": false,
            "expires_at": "2020-04-02 12:09:18",
            "last_event_at": "2014-02-01 12:43:56",
            "reason": "hard-bounce",
            "sender": "example.sender@mandrillapp.com",
            "subaccount": "example_subaccount"
        },
        "ts": 1454775960,
        "type": "blacklist"
    },
    {
        "action": "change",
        "reject": {
            "created_at": "2014-01-15 11:32:19",
            "detail": "Example detail",
            "email": "example.webhook@mandrillapp.com",
            "expired": false,
            "expires_at": "2020-04-02 12:09:18",
            "last_event_at": "2014-02-01 12:43:56",
            "reason": "hard-bounce",
            "sender": "example.sender@mandrillapp.com",
            "subaccount": "example_subaccount"
        },
        "ts": 1454775960,
        "type": "blacklist"
    },
    {
        "action": "remove",
        "reject": {
            "created_at": "2014-01-15 11:32:19",
            "detail": "Example detail",
            "email": "example.webhook@mandrillapp.com",
            "expired": false,
            "expires_at": "2020-04-02 12:09:18",
            "last_event_at": "2014-02-01 12:43:56",
            "reason": "hard-bounce",
            "sender": "example.sender@mandrillapp.com",
            "subaccount": "example_subaccount"
        },
        "ts": 1454775960,
        "type": "blacklist"
    },
    {
        "action": "add",
        "entry": {
            "created_at": "2014-01-15 12:03:19",
            "detail": "example details",
            "email": "example.webhook@mandrillapp.com"
        },
        "ts": 1454775960,
        "type": "whitelist"
    },
    {
        "action": "remove",
        "entry": {
            "created_at": "2014-01-15 12:03:19",
            "detail": "example details",
            "email": "example.webhook@mandrillapp.com"
        },
        "ts": 1454775960,
        "type": "whitelist"
    }
]

```

File: src/Mandrill.net/Model/WebHook/MandrillSyncEvent.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+    public class MandrillSyncEvent
+    {
+        private Dictionary<string, string> _reject;
+        
+        public MandrillSyncType Type { get; set; }
+        
+        public MandrillSyncAction Action { get; set; }
+
+        public Dictionary<string, string> Reject
```
Comment (feinoujc) [inline src/Mandrill.net/Model/WebHook/MandrillSyncEvent.cs]: What do you think if we use a typed object here?

something like this should work

```cs
    public class MandrillSyncEventInfo
    {
        [JsonConverter(typeof (IsoDateTimeConverter))]
        public DateTime CreatedAt { get; set; }
        public string Detail { get; set; }
        public string Email { get; set; }
        public bool? Expired { get; set; }
        [JsonConverter(typeof (IsoDateTimeConverter))]
        public DateTime? ExpiresAt { get; set; }
        [JsonConverter(typeof (IsoDateTimeConverter))]
        public DateTime? LastEventAt { get; set; }
        public string Reason { get; set; }
        public string Sender { get; set; }
        public string Subaccount { get; set; }
    }

```

**Comment by ChrisEby** [comment]
Good catch with the docs being off.  I'm going to re-work that & I'll add in the typed information instead of the Dict<string, string>.  That's a good point.  I can use that with my current side project to store these events.  I'll update & submit another pull request.  Thanks!

**Pull request closed without merging**